src/os/user/iterate_plan9.go

```diff
@@ -0,0 +1,101 @@
+package user
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+)
+
+// users and groups file location in plan9
+var (
+	usersFile = "/adm/users"
+)
+
+// userGroupIterator is a helper iterator function, which parses /adm/users
+func userGroupIterator(lineFn lineFunc) (err error) {
+	f, err := os.Open(usersFile)
+	if err != nil {
+		return fmt.Errorf("open users file: %w", err)
+	}
+	defer func() {
+		err = f.Close()
+	}()
+
+	_, err = readColonFile(f, lineFn, 3)
+	return
+}
+
+// matchPlan9UserGroup matches valid /adm/users line.
+// On plan9 /adm/users lines are both users and groups,
+//
+// idx should be set to -1 for iteration.
+//
+// Plan9 /adm/user line structure looks like this:
+// id:name:leader:members
+// sys:sys::glenda,mj <-- user/group without a leader, with 2 members glenda and mj
+// mj:mj:: <-- user/group without a leader, without members
+//
+// According to plan 9 users(6): ids are arbitrary text strings, typically the same as name.
+// In older Plan 9 file servers, ids are small decimal numbers.
+func matchPlan9UserGroup(value string, idx int, returnUser bool) lineFunc {
```

Comment:
consider not using a boolean for this; let's think about and review other approach(-es).